src/main/java/net/imglib2/parallel/Parallelization.java

```diff
@@ -0,0 +1,266 @@
+ * Martin Horn, Steffen Jaensch, Lee Kamentsky, Larry Lindsey, Melissa Linkert,
+ * Mark Longair, Brian Northan, Nick Perry, Curtis Rueden, Johannes Schindelin,
+ * Jean-Yves Tinevez and Michael Zinsmaier.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package net.imglib2.parallel;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * This class allows to configure an algorithm for parallelization.
+ * <p>
+ * The algorithm needs to use the {@link TaskExecutor} returned by
+ * {@link Parallelization#getTaskExecutor()} to implement the parallelization.
+ * Alternatively it can use {@link Parallelization#getExecutorService()}.
+ * But {@link TaskExecutor} is simpler and better suited for image precessing algorithms.
+ * <p>
+ * The algorithm can be executed singleThreaded, multiThreaded or by
+ * using a specified {@link ExecutorService} or {@link TaskExecutor}:
+ * <pre>
+ *     {@code
+ *     // Single Threaded call
+ *     Parallelization.singleThreaded().run( () -> myAlgorithm( image ) );
+ *
+ *     // Multi Threaded call
+ *     Parallelization.multiThreaded().run( () -> myAlgorithm( image ) );
+ *
+ *     // ExecutorService
+ *     Parallelization.withExecutor( executorService ).run( () -> myAlgorithm( image ) );
+ *
+ *     // Example Algorithm, that fills an image with ones.
+ *     public void myAlgorithm( RandomAccessibleInterval< IntType > image )
+ *     {
+ *         TaskExecutor taskExecutor = Parallelization.getTaskExecutor();
+ *         int numTasks = taskExecutor.suggestNumberOfTasks();
+ *         List< RandomAccessibleInterval< IntType > > chunks = splitImageIntoChunks( image, numTasks );
```

Comment:
Please add a note in here that it is the user's responsibility to create a method `splitImageIntoChunks`, otherwise people might be looking for that method in the imglib2 code base. Maybe also add a comment that chunking may be task specific, e.g. for separable distance transform.

Reply:
Lets add the `splitImageIntoChunks` method to imglib2. It would be great to have it!?

Reply:
I second that--having a utility method view an image as chunks would facilitate lots of things beyond parallel processing (but particularly parallel processing). Will need an out of bounds strategy that accesses the underlying ```RandomAccessibleInterval```, though.

Reply:
> I second that--having a utility method view an image as chunks would facilitate lots of things beyond parallel processing (but particularly parallel processing). Will need an out of bounds strategy that accesses the underlying `RandomAccessibleInterval`, though.

In my world, chunks almost always need context but I would almost never out-of-bounds-pad them if there is real context available.  I am therefore hesitant to think this method is generally useful for anything but per-pixel processing?  I always work with a set of plain Intervals generated by a customizable chunking method and passing along the original (out-of-bounds-padded) source and the Interval, so that the consuming method can do what is right with it (e.g. make an IntervalView).

Reply:
> I am therefore hesitant to think this method is generally useful for anything but per-pixel processing?

I tend to agree. But I'm open to suggestions!
Maybe a full-fledged generally useful `splitImageIntoChunks` is not possible, but certainly little helpers could be added to make special case implementations easier.

These discussions are related:
https://github.com/imglib/imglib2/issues/240
https://github.com/imglib/imglib2/pull/179
and maybe more...?

Reply:
Also related is `net.imglib.loop.IntervalChunks`, but it's not very flexible

Reply:
I agree with @axtimwalde and would like to add that there is the [`Grids`](https://github.com/imglib/imglib2-algorithm/blob/47cd6ed5c97cca4b316c92d4d3260086a335544d/src/main/java/net/imglib2/algorithm/util/Grids.java) utility class to generate lists of `Interval` from `Dimensions` and `blockSize`.

Reply:
> I am therefore hesitant to think this method is generally useful for anything but per-pixel processing?

That's probably correct. And LoopBuilder could be used for per-pixel processing instead. That makes the `splitImageIntoChunks` method rather useless. I will go with to original suggestion by @hanslovsky and change the javadoc.